internal/sql/pg/conn.go

```diff
@@ -264,3 +294,42 @@ func (p *Pool) BeginReadTx(ctx context.Context) (sql.Tx, error) {
 		accessMode: sql.ReadOnly,
 	}, nil
 }
+
+// Listen listens for notifications thrown by the custom `notify()` function.
+// `notify` is similar to kwil's `error` function, but is used for non-error
+// logging in the database. It returns a `done` function that cleans up
+// the listener. Only notifications raised in the passed transaction will be
+// used. The passed Tx must either be either a dbTx or a readTx; if it is a
+// anything else, it will return an error.
+func (p *Pool) Listen(ctx context.Context, tx sql.Tx, ch chan string) (func(), error) {
```

Comment:
This is really the only thing that raises an eyebrow.
When Listen gets used, we might find that it might as well be a method of the transactions, which contain a reference to the DB/Pool and thus the subscribers map.

Reply:
Yeah I considered that. Happy to make that change